bfc.C: fix for cling

veprbl opened the pull request:
cling needs to have libraries loaded before running a macro.

This disables the #include statements when ACLiC is not
used. Currently, cling can't parse those headers: there seems to be an
issue with how forward declarations interract with
`$clingAutoload$`. Fortunately, the `#pragma load` takes care of the
headers, like in ROOT5, so things just work.

veprbl committed:
bfc.C: fix for cling

cling needs to have libraries loaded before running a macro.

This disables the #include statements when ACLiC is not
used. Currently, cling can't parse those headers: there seems to be an
issue with how forward declarations interract with
"$clingAutoload$". Fortunately, the "#pragma load" takes care of the
headers, like in ROOT5, so things just work.

genevb reviewed (commented):
Hi, @veprbl . I understood @perevbnlgov was working on the transition of macros for ROOT6. Did this get discussed with him?

Thanks,
-Gene

veprbl committed:
bfc.C: add more libraries needed for ROOT 6.16

When tested on 6.24/06 the libraries seem to be loaded
automagically. On CI we use 6.16, and that seems to not work. Or it
could be that it uses RTLD_NOW. Let's try to add missing libraries
manually.

veprbl commented:
> Hi, @veprbl . I understood @perevbnlgov was working on the transition of macros for ROOT6. Did this get discussed with him?
> 
> Thanks, -Gene

I am only aware of what's discussed in threads https://github.com/star-bnl/star-sw/pulls?q=author%3Aperevbnlgov+ . There was no private discussion. If discussion is needed, it can happen here, I suppose.

klendathu2k commented on StRoot/macros/bfc.C:
```diff
@@ -9,7 +9,11 @@
 class StMessMgr;
-#if !defined(__CINT__) || defined(__MAKECINT__)
+
+#pragma cling load("StUtilities")
```
So... I did a search for documentation on this feature of cling... and only was able to find the load pragma in their github repo under tests...

https://github.com/root-project/cling

Regardless,... this appears to solve the problem, enabling the necessary libraries to be preloaded.  I *think* you could also achieve this by loading libraries in the standard rootlogon.C.  But this would be more general (and allows individual users to implement their own root logons).

veprbl replied:
Yes, rootlogon.C, might be possible, but, if it can work without the "secret sauce", I prefer that)

klendathu2k reviewed (approved):
LGTM

plexoos commented:
Last call for comments?
